### PR TITLE
Add css-individual-transform-properties blogpost (Fix #8244)

### LINKED ISSUE #8244: [CSS] Individual Properties for CSS Transforms [8/2/2022]
* Proposed title: Individual Properties for CSS Transforms
* Proposed by: bramus
* Related areas: CSS
* Content type: Reference (https://documentation.divio.com/reference/)
* Deadline: 8/2/2022
* Deadline explanation: Chrome 104 gets released then, but preferably this gets published earlier … it’s only a short post
* [Draft](https://docs.google.com/document/d/1PbhWmu3nqWPNcbXRopTCy5yU-WRSdnEGDSdLVRmJKWo/edit)
* Content summary: It’s gone unnoticed, but Individual Properties for CSS Transforms ship with Chrome 104. As it’s now stable across browsers, a post covering them seems necessary.

### PR DESCRIPTION
**Pull request by bramus**
Fixes: #8244 

This PR adds the `css-individual-transform-properties` blogpost, as described in https://github.com/GoogleChrome/web.dev/issues/8244

### CODE REVIEW

**Comment by bramus** [comment]
Had to bump up the version of `@mdn/browser-compat-data` used, as the compat table shown relies on https://github.com/mdn/browser-compat-data/pull/16816 which is included in its 5.1.4 release.

**Comment by rachelandrew** [comment]
@bramus can you do the compat data in a new PR? As this post will go out later than my new on the web platform post (which is when I normally update BCD). Also sometimes updating BCD breaks things so it's better alone.

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
+---
+layout: post
+title: Finer grained control over CSS transforms with individual transform properties
+authors:
+  - bramus
+  - dbaron
+description: >
+  Learn how you can use the individual translate, rotate, and scale CSS properties to approach transforms in an intuitive way.
+date: 2022-08-02
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:9]:
```suggestion
date: 2022-08-02
tags:
  - blog
  - newly-interoperable
```

Comment (rachelandrew) [reply src/site/content/en/blog/css-individual-transform-properties/index.md:9]: It won't show up on the blog without the blog tag.

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+---
+layout: post
+title: Finer grained control over CSS transforms with individual transform properties
+authors:
+  - bramus
+  - dbaron
+description: >
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:7]: We need a hero image and alt text. https://web.dev/handbook/use-media/

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  Learn how you can use the individual translate, rotate, and scale CSS properties to approach transforms in an intuitive way.
+date: 2022-08-02
+---
+
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:11]: Add the aside as in this post https://web.dev/transformstream/

I'll go through and tidy these up once we're not "newly interoperable" any more, as I can find them via the newly-interoperable tag.

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+```css
+.target {
+  translate: 50% 0;
+  rotate: 30deg;
+  scale: 1.2;
+}
+```
+
+## Order Matters
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:56]:
```suggestion
## Order matters
```

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+One key difference between the original CSS `transform` property and the new properties is the order in which the declared transformations get applied.
+
+With `transform`, the transformation functions get applied in the order they’re written–from left (outside) to right (inside).
+
+With the individual transformation properties, the order is not the order in which they are declared.  The order is always the same: first `translate` (outside), then `rotate`, and then `scale` (inside).
+
+That means both code snippets below yield the same result:
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:64]:
```suggestion
That means both of the following code snippets give the same result:
```

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+In both cases the targeted elements will first be translated by `50%` on the X-axis, then rotated by `30deg`, and finally scaled by `1.2`.
+
+If one of the individual transform properties are declared along with a `transform` property, then the individual transforms get applied first (`translate`, `rotate`, and then `scale`) with the `transform` last (inside). More details are in the spec that defines [how the transformation matrix should be calculated](https://www.w3.org/TR/css-transforms-2/#ctm).
+
+## Animations
+
+The main reason these properties were added is to make animations easier. Say you want to animate an element as follows:
+
+{% Img src="image/AeNB0cHNDkYPUYzDuv8gInYA9rY2/Pn0t3nrqfBg6JhmYFslm.svg", alt="Keyframes Graph", width="800", height="745" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:88]:
```suggestion
{% Img src="image/AeNB0cHNDkYPUYzDuv8gInYA9rY2/Pn0t3nrqfBg6JhmYFslm.svg", alt="Keyframes graph.", width="800", height="745" %}
```

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+### Using `transform`
+
+To implement this animation using `transform`, you’d have to calculate all in-between values for all defined transformations, and include those in each keyframe. For example, to do a rotation at the 10% mark, the values for the other transformations must be calculated as well, because the `transform` property needs all of them.
+
+{% Img src="image/AeNB0cHNDkYPUYzDuv8gInYA9rY2/AF5n6UlhcuQ5UNKemX8M.svg", alt="Keyframes Graph with intermediate values calculated", width="800", height="745" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:94]:
```suggestion
{% Img src="image/AeNB0cHNDkYPUYzDuv8gInYA9rY2/AF5n6UlhcuQ5UNKemX8M.svg", alt="Keyframes graph with intermediate values calculated.", width="800", height="745" %}
```

File: src/site/content/en/blog/css-individual-transform-properties/index.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+  100% { transform: translateX(100%) rotate(360deg); }
+}
+
+.target {
+  animation: anim 2s;
+  animation-fill-mode: forwards;
+}
+```
+
+{% Codepen {
```
Comment (rachelandrew) [inline src/site/content/en/blog/css-individual-transform-properties/index.md:114]: I need to send you the web.dev CodePen login details, as it's better they are in there as then anyone can update them if needed. Just fork yours into the account.

**Review by rachelandrew — CHANGES_REQUESTED** [review]
Changes requested inline, also as mentioned in a comment can we make the BCD update as a separate PR.

**Comment by bramus** [comment]
Added the requested changes.

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**